File: backend/src/main/java/com/springboot/library/entity/Book.java
package com.springboot.library.entity;


import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Lob;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.springframework.lang.NonNull;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name="book")
public class Book{
	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="book_id")
	private Long bookId;
	
	@NonNull
	private String bookName;
	
	private String bookImg;
	
	@NonNull
	@Lob
	private String bookDescription;
	private String authorName;
	
	private Date publishedDate;
	
	@JsonIgnore
	@OneToMany(mappedBy="book")
	public List<UserBooks> userbooks;
}

File: library/src/app/model/Book.ts
import { User } from "./User";

export class Book{
    public bookId : number;
    public bookName : string;
    public bookImg : string;
    public bookDescription : string;
    public authorName : string;
    public publishedDate : Date;
    public userList : User[];

    constructor(bookId : number, bookName : string, bookImg : string, bookDescription : string, authorName : string, publishedDate : Date, userList : User[]) {
        this.bookId = bookId;
        this.bookName = bookName;
        this.bookImg = bookImg;
        this.bookDescription = bookDescription;
        this.authorName = authorName;
        this.publishedDate = publishedDate
        this.userList = userList;
        
    }

}